flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -897,7 +897,7 @@ class TaskManager(
         config.timeout,
         libCache,
         fileCache,
-        runtimeInfo)
+        new TaskRuntimeInfo(hostname, taskManagerConfig, tdd.getAttemptNumber))
```

Comment:
Yes. The addition of distributed cache removes the need for multiple constructors for `RuntimeContext`s. Since providing access to runtime information needed changing the constructors, I deemed it better to work with what would be the only needed constructors after merging this. 
I can revert this commit and open a separate PR for the *other* three issues if necessary.